### PR TITLE
Make runtime of findPlace deterministic

### PR DESCRIPTION
**Pull request by J3RN**
Description
-----------

Theoretically, `findPlace` in it's current implementation could run
indefinitely supposing that `randomPlace` kept returning "Thai." Of
course, this would be an exceptionally rare occurance whose sheer
implausibility would be compounded every iteration. That said, I've
got a sheet of paper in a box somewhere in my kitchen that says I'm
supposed to take this very seriously.

The runtime of the new algorithm is O(n) where 'n' is the length of the
list of places, given the following assumptions (God help us):

- Allocating an array of length 'n' is an O(n) operation.
- Calling `filter` on an array of length 'n' is an O(n) operation,
  assuming the lambda passed runs in constant time.
- `includes` runs in constant time (this is obviously not true, but
  I'm tired and I want it to be).
- `Math.floor`, `Math.random`, and `length` all also run in constant
  time. There's no very good reason why any of these should *not* be
  the case, but knowing JavaScript, I'm going to take all available
  precautions.
- I am of sound mind and body. Or at the very least the former, which
  is coincidentally the lesser likely of the two to actually be the
  case.

This does obviate the necessity of having both `findPlace` and
`randomPlace`, though one could argue that that has always been the
case.

It also contains some function calls that may not be supported by
older browsers or negligent fathers. They are:

- [String.prototype.includes()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes#Browser_compatibility)
  (Not supported by Internet Explorer; but then again what is?)
- [Array.prototype.filter()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter#Browser_compatibility)
  (Not supported by Internet Explorer < 9)

Testing
-------

Functions that are random are inherently difficult to test. However, I
have dedicated an hour of computing time to clicking the "Where do I
want to Eat" button incessantly with the "Exclude Thai" box checked,
and did not see anything resembling "Thai" be offered as a
suggestion. If it's good enough for the Four Color Theorem, it's good
enough for me.

Conclusion
----------

In summary, I suggest that the feasibility of somehow tagging each
restaurant entry with its cuisine be evaluated. I have discovered that
it is possible—though, in practice, extremely uncommon—to have a Thai
restaurant without the word "Thai" in the name.

### CODE REVIEW

**Comment by mwagz** [comment]
Looks good. I'll take it.
